src/reducers/ComparisonSlice.ts

```diff
@@ -33,6 +43,56 @@ const comparison = createSlice({
   },
 });
 
+export const formatDownloadData = (
+  data: Record<string, ResultObject[]>,
+): object[] => {
+  const groupNames = Object.keys(data);
+  const transformedGroups: object[] = [];
+
+  groupNames.forEach((groupName) => {
+    const groupedResults = data[groupName].reduce(
+      (grouped: GroupedResults, result: ResultObject) => {
+        if (!grouped[result.header_name]) {
+          grouped[result.header_name] = [];
+        }
+        grouped[result.header_name].push(result);
+        return grouped;
+      },
+      {},
+    );
+
+    const transformedGroupEntries = Object.keys(groupedResults).map(
+      (header_name) => ({
+        [`${header_name} ${truncateHash(
+          groupedResults[header_name][0].new_rev,
+        )}`]: groupedResults[header_name],
```

Comment:
nit: can you please compute the key in a separate line, so that the resulting structure is easier to understand?